core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -428,7 +422,7 @@ private object RTS {
                         val defects = status.get.defects
 
                         curIo = null
-                        result = ExitResult.Failed(error, defects)
+                        result = ExitResult.Terminated(Cause.failure(error, defects))
```

Comment:
We may want to add utility functions for that, like `ExitResult.failed(error, defects)`.